scripts/run-plugin-nuts.js

```diff
@@ -0,0 +1,183 @@
+    return manualRepoOverrides[module];
+  }
+  const repoUrl = new URL(results.toString().trim());
+  const [, org, repo] = repoUrl.pathname.split('/');
+  return { org, repo };
+};
+
+const orgSlug = (org, repo) => {
+  return `github/${org}`;
+};
+const projectSlug = (org, repo) => {
+  return `${orgSlug(org)}/${repo}`;
+};
+
+/**
+ * Triggers the just-nuts workflow for the named org/repo/branch
+ * @param sfdxVersion
+ * @param org
+ * @param repo
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsForProject = async (sfdxVersion, org, repo, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      sfdx_version: sfdxVersion,
+    },
+  };
+  return circle(`${circleciBaseUrl}project/${projectSlug(org, repo)}/pipeline`, { method: 'POST', json: body });
+};
+
+/**
+ * Triggers the nuts monitoring workflow
+ * @param jobData
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsMonitor = async (jobData, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      nuts_job_data: JSON.stringify(jobData),
+    },
+  };
+  return circle(`${circleciBaseUrl}project/${projectSlug('salesforcecli', 'sfdx-cli')}/pipeline`, {
+    method: 'POST',
+    json: body,
+  });
+};
+/**
+ * Makes a request to named URL. Ensures that auth token is applied.
+ * @param url
+ * @param options
+ * @returns {Promise<any>}
+ */
+const circle = async (url, options = {}) => {
+  const defaultOptions = {
+    method: 'GET',
+    headers: { 'Circle-Token': process.env.CIRCLECI_API_TOKEN, responseType: 'json', resolveBodyOnly: true },
+  };
+  options = { ...defaultOptions, ...options };
+  const response = await (async () => {
+    try {
+      return await got(url, options);
+    } catch (error) {
+      console.log(error);
+      throw new Error(error && error.response?.body ? error.response.body : error);
+    }
+  })();
+  return JSON.parse(response.body);
+};
+
+/**
+ * Examines a lit of plugins to determine which have NUT tests
+ * @param plugins
+ * @returns {*}
+ */
+const qualifyPluginsWithNonUnitTests = (plugins) => {
+  return (
+    plugins
+      // find plugins that may have NUT tests
+      .filter((plugin) => plugin === 'salesforcedx' || plugin.startsWith('@salesforce'))
+      // handle sub-aggregate plugins by fetching that plugins dependencies
+      .map((plugin) => {
+        if (plugin === 'salesforcedx') {
+          const dxDependencies = Reflect.ownKeys(getNpmDependencies(plugin));
+          return dxDependencies;
+        } else {
+          return plugin;
+        }
+      })
+      // flatten arrays
+      .reduce((a, b) => {
+        return a.concat(b);
+      }, [])
+      // find plugins that have a script named 'test:nuts'
+      .filter((plugin) => hasTestNuts(plugin))
```

Comment:
This is a super edge case. We run these nuts on the `latest-rc`. If a plugin published a `latest-rc` tag which introduced test:nuts, this wouldn't pick it up because it is checking that script on the latest tag.

I don't think we really need to account for this since it will rarely happen and it is only for the first week they introduce it, all other weeks will get picked up. SO I think it is fine.

Reply:
@amphro  Since the goal of just-nuts is to force plugin NUT tests to reply on the installed `sfdx`, rather than `bin/run`, so I should not depend on tags applied to plugins, since those advance at a different rate than sfdx-cli. In reality, the should be checking the absolute version of plugins installed with sfdx-cli to determine if test:nuts are available and possibly change the check for plugins to checkout source for the same version found in sfdx-cli installation, just to make sure the tests run aligned with sfdx-cli version.

Reply:
Checking the absolute version and running test from the commit has of that version would definitely be better, but maybe lets do that in a follow up? Unless you think it would be a pretty easy change to do that. It still feels like an edge case so I don't want this story to keep getting bigger.